cupyx/fallback_mode/utils.py

```diff
@@ -33,11 +42,108 @@ def _call_numpy(func, args, kwargs):
     Returns:
         Result after calling func.
     """
+    args, kwargs = data_transfer._get_cupy_ndarray(ndarray, args, kwargs)
 
     numpy_args, numpy_kwargs = data_transfer._get_numpy_args(args, kwargs)
 
     numpy_res = func(*numpy_args, **numpy_kwargs)
 
     cupy_res = data_transfer._get_cupy_result(numpy_res)
 
-    return cupy_res
+    return data_transfer._get_fallback_ndarray(ndarray, cupy_res)
+
+
+def make_method(name):
+    def method(self, *args, **kwargs):
+
+        args, kwargs = data_transfer._get_cupy_ndarray(ndarray, args, kwargs)
+
+        cupy_method = getattr(cp.ndarray, name)
+
+        res = cupy_method(self._array, *args, **kwargs)
+
+        return data_transfer._get_fallback_ndarray(ndarray, res)
+
+    return method
+
+
+def _create_magic_methods():
+
+    _magic_methods = [
+
+        # Comparison operators:
+        '__richcmp__',
+
+        # Truth value of an array (bool):
+        '__nonzero__',
+
+        # Unary operations:
+        '__neg__', '__pos__', '__abs__', '__invert__',
+
+        # Arithmetic:
+        '__add__', '__sub__', '__mul__', '__matmul__', '__div__',
+        '__truediv__', '__floordiv__', '__mod__', '__divmod__', '__pow__',
+        '__lshift__', '__rshift__', '__and__', '__or__', '__xor__',
+
+        # Arithmetic, in-place:
+        '__iadd__', '__isub__', '__imul__', '__idiv__', '__itruediv__',
+        '__ifloordiv__', '__imod__', '__ipow__', '__ilshift__', '__irshift__',
+        '__iand__', '__ior__', '__ixor__',
+
+        # For standard library functions:
+        '__copy__', '__deepcopy__', '__reduce__',
+
+        # Container customization:
+        '__iter__', '__len__', '__getitem__', '__setitem__',
+
+        # Conversion:
+        '__int__', '__float__', '__complex__', '__oct__', '__hex__',
+
+        # String representations:
+        '__repr__', '__str__'
+    ]
+
+    for method in _magic_methods:
+        setattr(ndarray, method, make_method(method))
+
+
+class ndarray:
+
+    def __init__(self, array):
+        self._array = array
+        self.func = None
+        self._numpy_func = False
+
+    def __getattr__(self, attr):
+
+        self._numpy_func = False
+
+        self.func = getattr(cp.ndarray, attr, None)
```

Comment:
Do not pass this function to `self.__call` via `self.func`. This bug will be reproduced as follows:
```py
>>> from cupyx.fallback_mode import numpy
>>> x = numpy.array([1, 2, 3])
>>> sum = x.sum
>>> sum()  # x.func is x.sum
array(6)
>>> x.shape  # Overwrite x.func
(3,)
>>> sum()  # x.func is x.shape
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/imanishi/cupy/cupyx/fallback_mode/utils.py", line 137, in __call
    return _call_cupy(self.func, args, kwargs)
  File "/home/imanishi/cupy/cupyx/fallback_mode/utils.py", line 28, in _call_cupy
    res = func(*args, **kwargs)
TypeError: 'getset_descriptor' object is not callable
```

Reply:
How about returning `_RecursiveAttr` object?

Reply:
That's a good idea. Will need to make some adjustments though.

Reply:
Done